src/types/block.rs

```diff
@@ -158,7 +158,11 @@ impl Serialize for BlockId {
         S: Serializer,
     {
         match *self {
-            BlockId::Hash(ref x) => serializer.serialize_str(&format!("0x{}", x)),
+            BlockId::Hash(ref x) => {
```

Comment:
Initially I thought this is going to break some of the existing APIs using `BlockId`, but it's not the case. In fact it seems that the fact that `Serialize` is implemented for `BlockId` is not used anywhere in the `rust-web3` codebase internally, so it's okay to change it.

Reply:
Exactly.